provisioning/miniprov/hare_mp/main.py

```diff
@@ -144,6 +144,15 @@ def get_server_type(url: str) -> str:
         logging.error('Cannot get server type (%s)', error)
         return 'unknown'
 
+def is_stage_required(url: str) -> str:
+    try:
+        provider = ConfStoreProvider(url)
+        machine_id = provider.get_machine_id()
+        pod_type = provider.get(f'node>{machine_id}>type')
+        return pod_type == 'server_node'
+    except Exception as error:
```

Comment:
It doesn't make much sense to return 'unknown' string: 
1. It is very easy to forget to check the return code in the calling code
2. The function name gives a clue that the return type will be boolean (i.e. the return value is understood either as yes or no). Thus if somebody uses the function like this 
```python
if is_stage_required(url):
    do_smth()
``` 
then `do_smth()` *will be executed* in case of an exception. Reason: non-empty string (like `'unknown'` is) evaluates to True.

Suggestion:
Raise an exception when something unexpected happens, don't rely on return codes. In this case the code will be kept clean (no paranoid "if error" checks) but on the other hand if an error happens, it will not be swallowed because of a human mistake.